tests/consistency.rs

```diff
@@ -39,6 +39,100 @@ fn compare_birds() {
     compare_audio(&audio);
 }
 
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_1() {
+    let audio = Path::new("tests/audio/sample_1.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_2() {
+    let audio = Path::new("tests/audio/sample_2.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_3() {
+    let audio = Path::new("tests/audio/sample_3.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_4() {
+    let audio = Path::new("tests/audio/sample_4.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+/// When is_speaking is true, get_current_speech should return everything starting from
+/// speech start time until the very end, including the unprocessed samples.
+fn should_include_unprocessed_when_is_speaking(audio: &Path, chunk_ms: usize) {
+    let config = VadConfig::default();
+    let mut vad = VadSession::new(config).unwrap();
+
+    // Read audio samples.
+    let step = if config.sample_rate == 16000 {
+        1
+    } else {
+        2 // Other sample rates are invalid so we'll just work on less data
+    };
+    let samples: Vec<f32> = WavReader::open(audio)
+        .unwrap()
+        .into_samples()
+        .step_by(step)
+        .map(|x| {
+            let modified = x.unwrap_or(0i16) as f32 / (i16::MAX as f32);
+            modified.clamp(-1.0, 1.0)
+        })
+        .collect();
+
+    // Send each chunk to vad for inference.
+    let chunk_size = (config.sample_rate * chunk_ms) / 1000;
+    let num_chunks = samples.len() / chunk_size;
+    let mut last_speech_start_ms = 0;
+    for i in 0..num_chunks {
+        let start = i * chunk_size;
+        let end = if i < num_chunks - 1 {
+            start + chunk_size
+        } else {
+            samples.len()
+        };
+
+        let transitions = vad.process(&samples[start..end]).unwrap();
+        // Update last_speech_start_ms
+        for transition in transitions {
+            if let VadTransition::SpeechStart { timestamp_ms } = transition {
+                last_speech_start_ms = timestamp_ms;
+            }
+        }
+
+        // The actual test logic.
+        if vad.is_speaking() {
```

Comment:
So follow on from the last review comment. There's no point testing with chunk size of 30 because there's no remainder then so nothing will be in unprocessed. Maybe it would be good to try a wackier size like 47 (prime number and over the window size). 

Also, to ensure we don't end up with useless tests that never hit the branch you should probably add a bool to the test that goes from false to true when this condition is entered and at the end `assert!(conditions_tested, "test was useless");` 

Adding a `#[doc(hidden)]` for the current unprocessed duration as well and asserting that goes >0 during the test would also be a nice addition that makes it even more robust.